File: solutions/csharp/protein-translation/ProteinTranslation.cs
using System;
using System.Linq;
public static class ProteinTranslation
{
    public static string[] Proteins(string strand)
    {
        return strand
            .Chunk(3)
            .Select(x => new string(x))
            .Select(Convert)
            .TakeWhile(x => x is not null)
            .ToArray();
    }

    private static string Convert(string tri) {
        return tri switch {
            "AUG" => "Methionine",
            "UUU" or "UUC" => "Phenylalanine",
            "UUA" or "UUG" => "Leucine",
            "UCU" or "UCC" or "UCA" or "UCG" => "Serine",
            "UAU" or "UAC" => "Tyrosine",
            "UGU" or "UGC" => "Cysteine",
            "UGG" => "Tryptophan",
            _ => null,
        };
    }
}

File: solutions/fsharp/protein-translation/ProteinTranslation.fs
module ProteinTranslation

let convert = function
| "AUG" -> "Methionine"
| "UUU" | "UUC" -> "Phenylalanine"
| "UUA" | "UUG" -> "Leucine"
| "UCU" | "UCC" | "UCA" | "UCG" -> "Serine"
| "UAU" | "UAC" -> "Tyrosine"
| "UGU" | "UGC" -> "Cysteine"
| "UGG" -> "Tryptophan"
| _ -> "STOP"

let proteins: string -> string list =
    Seq.toArray
    >> Array.chunkBySize 3
    >> Array.map System.String
    >> Array.map convert
    >> Array.takeWhile ((<>) "STOP")
    >> Array.toList